ruboty-channel-gacha.rb

```diff
@@ -12,16 +12,25 @@ class ChannelGacha < Base
       )
 
       def channel_gacha(message)
-        message.tap(&set_option).reply format(selected_channel)
+        set_option(message)
+        message.reply messages.join("\n")
       end
 
       private
 
-      def set_option
-        -> (message) { @option = message.match_data['option'] }
+      def set_option(message)
+        @option = message.match_data['option']
       end
 
-      def format(channel)
+      def messages
+        [pre_message, main_message(selected_channel)].compact
+      end
+
+      def pre_message
+        @option.split('=', 2).tap(&:shift) if with_pre_message?
```

Comment:
```suggestion
        @option.split('=', 2)[1] if with_pre_message?
```

ここではsplitした時の後ろの要素がほしいだけなので、tapを使うよりも`[]`を使って配列にアクセスした方がわかりやすいと思ったのですがどうでしょうか？

Reply:
最初に実装した時点では `split` の第2引数で分割数の上限を指定できるのを知らなくて、`message_content` に `=` が入っていたときを考えて現在の実装になっていました。
今はそれは考慮する必要がないのでご指摘の通り `[]` で指定したほうがいいですね 🙏